examples/thermostat/silabs/src/SensorManager.cpp

```diff
@@ -115,12 +115,20 @@ void SensorManager::SensorTimerEventHandler(void * arg)
 
     if ((temperature >= (lastTemperature + kMinTemperatureDelta)) || temperature <= (lastTemperature - kMinTemperatureDelta))
     {
-        lastTemperature = temperature;
-        PlatformMgr().LockChipStack();
+        lastTemperature            = temperature;
+        AttributeUpdateInfo * data = chip::Platform::New<AttributeUpdateInfo>();
+        data->endPoint             = kThermostatEndpoint;
+        data->temperature          = temperature;
         // The SensorMagager shouldn't be aware of the Endpoint ID TODO Fix this.
         // TODO Per Spec we should also apply the Offset stored in the same cluster before saving the temp
 
-        app::Clusters::Thermostat::Attributes::LocalTemperature::Set(kThermostatEndpoint, temperature);
-        PlatformMgr().UnlockChipStack();
+        chip::DeviceLayer::PlatformMgr().ScheduleWork(UpdateTemperatureAttribute, reinterpret_cast<intptr_t>(data));
```

Comment:
that can work... But it looks like we are jumping through some hoops to get the attribute update done on the chip task. Wouldn't it be easier to let the appTask of this sensor app handle what is done in SensorTimerEventHandler?

Keep the timer call back if you want and set in the app task queue this function that "calculate" the temperature with the lock in place.
This way the lock occurs in the app task context and not the timer task context